### PR TITLE
leader proxy back port to 1.4

### PR DESCRIPTION
**Pull request by kensipe**
This is a revert followed by 2 cherry picks that were reasonably clean.

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/api/LeaderProxyFilter.scala [hunk 1]
```diff
@@ -9,17 +9,22 @@ import javax.servlet._
 import javax.servlet.http.{ HttpServletRequest, HttpServletResponse }
 
 import akka.Done
+import akka.http.scaladsl.model.StatusCodes._
 import com.google.inject.Inject
 import mesosphere.chaos.http.HttpConf
 import mesosphere.marathon.core.election.ElectionService
 import mesosphere.marathon.io.IO
+<<<<<<< HEAD
```
Comment (timcharper) [inline src/main/scala/mesosphere/marathon/api/LeaderProxyFilter.scala]: looks like conflict markers were committed?

Comment (kensipe) [reply src/main/scala/mesosphere/marathon/api/LeaderProxyFilter.scala]: that is odd..  looking

File: src/main/scala/mesosphere/marathon/api/LeaderProxyFilter.scala [hunk 2]
```diff
@@ -275,20 +277,25 @@ class JavaUrlConnectionRequestForwarder @Inject() (
...
         try {
-          copyRequestToConnection(leaderConnection, request)
-          copyConnectionResponse(
-            response
-          )(
+          copyRequestToConnection(leaderConnection, request) match {
+            case Failure(ex: ConnectException) =>
+              log.error(ERROR_STATUS_CONNECTION_REFUSED, ex)
+              response.sendError(BadGateway.intValue, ERROR_STATUS_CONNECTION_REFUSED)
+              return
```
Comment (timcharper) [inline src/main/scala/mesosphere/marathon/api/LeaderProxyFilter.scala]: Can we rework this code to remove the need for this return statement? The multiple finally statements, plus the return, seem to make the control of this method pretty difficult to follow.

**Comment by kensipe** [comment]
This is currently WIP...   we have 1 more change to make to master to cherry pick to this PR

**Comment by jeschkies** [comment]
@kensipe can we close this diff?

**Comment by kensipe** [comment]
I think we still have the need for the backport... just need to prioritize it

**Comment by jeschkies** [comment]
It's been another two months. What's missing for it to merge?

**Comment by jeschkies** [comment]
@kensipe what is the state of this PR?

**Comment by timcharper** [comment]
Closing. @kensipe please re-open if this is worth pursuing further.

**Pull request closed without merging**